src/Psd/PsdParseColorModeDataSection.cpp

```diff
@@ -28,7 +50,13 @@ void DestroyColorModeDataSection(ColorModeDataSection*& section, Allocator* allo
 	PSD_ASSERT_NOT_NULL(section);
 	PSD_ASSERT_NOT_NULL(allocator);
 
-	memoryUtil::Free(allocator, section);
+	if (section) {

```

Comment:
There is no need to check for a valid section here, since the code above already asserts on this.
Calling DestroyColorModeDataSection() without a valid section is considered a user error.